packages/grafana-data/src/types/annotations.ts

```diff
@@ -86,24 +90,35 @@ export type AnnotationEventMappings = Partial<Record<keyof AnnotationEvent, Anno
 export interface AnnotationSupport<TQuery extends DataQuery = DataQuery, TAnno = AnnotationQuery<TQuery>> {
   /**
    * This hook lets you manipulate any existing stored values before running them though the processor.
-   * This is particularly helpful when dealing with migrating old formats.  ie query as a string vs object
+   * This is particularly helpful when dealing with migrating old formats.  ie query as a string vs object.
    */
-  prepareAnnotation?(json: any): TAnno;
+  prepareAnnotation?(json: AnnotationQuery<TQuery>): AnnotationQuery<TQuery>;
 
   /**
    * Convert the stored JSON model to a standard datasource query object.
    * This query will be executed in the datasource and the results converted into events.
    * Returning an undefined result will quietly skip query execution
    */
-  prepareQuery?(anno: TAnno): TQuery | undefined;
+  prepareQuery?(anno: AnnotationQuery<TQuery>): TQuery | undefined;
```

Comment:
Any reason for this change?  

The optional `TAnno = AnnotationQuery<TQuery>` is equivolent, but has some custom implementations in enterprise (and external) plugins

Reply:
Can you point to some code where this is needed. Imho being able to still supply anything as `TAnno` does not seem great. If we wanted to make this better I assumed we wanted a stricter typing. Having `TAnno` without any constraint seems like we would be back to allowing anything as annotation and having `TAnno extends AnnotationQuery` also does not seem beneficial as `AnnotationQuery` already is object that can have any key.